File: backend/apps/comments/realtime/contracts.py
from enum import StrEnum


class CommentTopic(StrEnum):
    COMMENTS = "comments"


class CommentOperation(StrEnum):
    SUBSCRIBE = "subscribe"
    CREATE = "comments.create"
    REPLY = "comments.reply"


class CommentEvent(StrEnum):
    CREATED = "comment.created"
    VOTED = "comment.voted"
    SEARCH_INDEXED = "search.indexed"


class CommentKind(StrEnum):
    ROOT = "root"
    REPLY = "reply"


class SocketMessageType(StrEnum):
    RESPONSE = "response"
    ERROR = "error"
    EVENT = "event"
    SUBSCRIBED = "subscribed"

File: frontend/src/realtime/contracts.ts
export enum CommentTopic {
  Comments = "comments",
}

export enum CommentOperation {
  Subscribe = "subscribe",
  Create = "comments.create",
  Reply = "comments.reply",
}

export enum CommentEvent {
  Created = "comment.created",
  Voted = "comment.voted",
}

export enum SocketMessageType {
  Response = "response",
  Error = "error",
  Event = "event",
  Subscribed = "subscribed",
}

export type SocketStatus = "connecting" | "open" | "closed";
